### DISCUSSION TITLE
I am not a Rust developer, but it seems similar to Livewire?

### DISCUSSION THREAD

**Discussion by PR4M** [leptos-rs/leptos#2418] [General]
Just curious, any PHP/Laravel developer here that ever-used Livewire? In my quick view, it seems quite similar to Livewire.

**Comment by gbj** [comment]
It is much, much, much more similar to something like React with NextJS or Remix, Svelte with SvelteKit, Vue with Nuxt, Solid with SolidStart, etc.

Superficially, it is similar to Livewire in the sense that it allows you to write an interactive web application in a language that is not JavaScript.

Mechanically (i.e., in terms of how it works) it is very different, with some real advantages.

Taking the description of LiveWire's flow from their home page:
>  Livewire renders the initial component output with the page (like a Blade include). This way, it's SEO friendly.

Yes, we do this.

> When an interaction occurs, Livewire makes an AJAX request to the server with the updated data.
> The server re-renders the component and responds with the new HTML.
> Livewire then intelligently mutates DOM according to the things that changed.

No, we don't do this, and it's much better.

Instead, we
- Render the initial page as HTML on the server
- Add interactivity to that page, in the browser, so that 
- When an interaction occurs, it is handled in the browser, only going to the server if needed for new data

i.e., with LiveWire, if you click a button, the page makes a new request to the server, loads a whole new HTML page, diffs it against the old HTML page, and then updates the single part that needs to change. This is a much slower and less efficient model for interactivity than what Leptos or all the JS frameworks listed above do.

**Reply by PR4M** [reply]
Thank you for such detail explanation! 

I will surely comeback and try Leptos when I'm using Rust. 

Have a great weekend, devs.

**Comment by PR4M** [comment]
Replied by the maintainer itself. Kudos to them.
